### PR TITLE
Pipe: release resources of pipe event that have been GCed but the reference count is not zero by PhantomReference

### PR DESCRIPTION
**Pull request by VGalaxies**
Considering that when the event is garbage collected, the file resources it holds will not automatically end their lifecycle, we adopt a reference counting maintenance + safety net fallback method (releasing resources later is better than never releasing them).

### CODE REVIEW

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+  public PipePhantomReferenceManager() {
+    // Do nothing now.
+  }
+
+  public int getPhantomReferenceCount() {
+    return PIPE_EVENT_PHANTOM_REFERENCES.size();
+  }
+
+  protected void gcHook() {
```
Comment (SteveYurongSu) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:52]: Use a seperated thread pool (extends PipePeriodicalJobExecutor?) to run this method periodically?

Comment (SteveYurongSu) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:52]: For one thread, what is the max number of events per second can be checked?

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+    private final String coreReportMessageSnapshot;
+    private PipeEventResource resource;
+
+    private PipeEventPhantomReference(
+        final EnrichedEvent event,
+        final PipeEventResource resource,
+        final ReferenceQueue<? super EnrichedEvent> queue) {
+      super(event, queue);
+      this.coreReportMessageSnapshot = event.coreReportMessage();
```
Comment (SteveYurongSu) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:87]: Calling coreReportMessage is time consuming ...

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java (hunk 1 above)
Comment (SteveYurongSu) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:52]: We have to consider using a multi-threaded pool to run this method if necessary (maybe we can add a conf parma for this?)

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      this.referenceCount = referenceCount;
+    }
+
+    private void clearReferenceCount(final String holderMessage) {
+      if (isReleased.get()) {
+        return;
+      }
+
+      if (referenceCount.get() >= 1) {
+        LOGGER.info("finalize resource for event with holder message {}", holderMessage);
```
Comment (SteveYurongSu) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:127]:
```suggestion
        LOGGER.error("finalize resource for event with holder message {}", holderMessage);
```
"RESOURCE LEAK DETECTED: {}....", class name...

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java (hunk 1 above)
Comment (SteveYurongSu) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:52]: add a check here: if this run is enabled by conf

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonDescriptor.java [hunk 4]
```diff
@@ -604,6 +604,17 @@ private void loadPipeProps(Properties properties) {
...
+
+    config.setPipeEventReferenceTrackingEnabled(
+        Boolean.parseBoolean(
+            properties.getProperty(
+                "pipe_event_reference_tracking_enabled",
+                String.valueOf(config.getPipeEventReferenceTrackingEnabled()))));
+    config.setPipeEventReferenceEliminateIntervalSeconds(
+        Long.parseLong(
+            properties.getProperty(
+                "pipe_event_reference_eliminate_interval_seconds",
```
Comment (SteveYurongSu) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonDescriptor.java:616]: Can be shorter by default?

Comment (VGalaxies) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/conf/CommonDescriptor.java:616]: no other ideas for now...

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java (hunk 1 above)
Comment (VGalaxies) [reply iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/resource/ref/PipePhantomReferenceManager.java:53]: consider optimizing in the next PR...

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**